apps/s_client.c

```diff
@@ -798,6 +801,31 @@ static int new_session_cb(SSL *S, SSL_SESSION *sess)
     return 0;
 }
 
+static void keylog_callback(const SSL *ssl, const char *line)
+{
+    BIO *keylog = BIO_new_file(keylog_file, "a");
+    char buffer[256];
```

Comment:
Not sure what you exactly refer to by serialization, the whole line is one unit of information which must be terminated by a newline. If a previous "line" did not terminate correctly, then the next write would end up on the previous line.

I remember (but cannot find a reference) that small writes (at least on Linux?) are "atomic", either it succeeds or you get EINTR and the write could fail. Will try to get find the reference when I am a bit more awake.